triplestore/src/triplestore/network_utils.py

```diff
@@ -30,3 +50,77 @@ def detect_host_url(port: int, path: str = "", fallback: str | None = None) -> s
 
 def detect_graphdb_url() -> str:
     return detect_host_url(7200)
+
+
+def validate_config(user_config: Mapping[str, Any], *, required_keys: Iterable[str], optional_defaults: Mapping[str, Any] | None,
+                    alias_map: Mapping[str, Any] | None, backend_name: str = "backend") -> dict[str, Any]:
+    """
+    Validate and normalize a backend configuration dictionary.
+
+    This function ensures that:
+    1. All required keys are present (after resolving aliases).
+    2. Optional keys are filled in with defaults if not provided.
+    3. Unknown keys trigger a warning message (but are preserved in the result).
+
+    Parameters
+    ----------
+    user_config : Mapping[str, Any]
+        The configuration dictionary provided by the user.
+    required_keys : Iterable[str]
+        Keys that must always be present in the final configuration.
+    optional_defaults : Mapping[str, Any], optional
+        Optional keys with their default values if missing.
+    alias_map : Mapping[str, str], optional
+        Mapping of alias → canonical key names.
+    backend_name : str, default="backend"
+        Name of the backend, used in error/warning messages.
+
+    Returns
+    -------
+    dict[str, Any]
+        A normalized configuration dictionary containing:
+        - All required keys,
+        - All optional keys (with user or default values),
+        - All provided aliases converted to canonical keys,
+        - Any unknown keys (with a warning).
+
+    Raises
+    ------
+    ValueError
+        If one or more required keys are missing.
+    """
+
+    if optional_defaults is None:
+        optional_defaults = {}
+    if alias_map is None:
+        alias_map = {}
+
+    normalized_config: dict[str, Any] = {}
+    for key, value in user_config.items():
+        canonical_key = alias_map.get(key, key)
+        normalized_config[canonical_key] = value
+
+    missing_keys = [k for k in required_keys if k not in normalized_config]
+    if missing_keys:
+        msg = (
+            f"[{backend_name}] Configuration error: Missing required config keys for: '"
+            f"{', '.join(missing_keys)}'"
+        )
+        raise ValueError(msg)

```

Comment:
```suggestion
        raise TriplestoreMissingConfigValue(msg)

```
and define it properly